src/base/amo.hh

```diff
@@ -126,13 +126,19 @@ class AtomicGenericPair3Op : public TypedAtomicOpFunctor<T>
     std::function<void(T*, std::array<T, 2>&, std::array<T, 2>)> op;
 };
 
+// IMPORTANT: The following operations assume the guest is little-endian.
```

Comment:
even if we decide to drop everything else, we should probably keep this

Reply:
If we leave this (and drop the other changes), then the comment should read `IMPORTANT: The following code assumes the guest and host have the same endianness. If this code is used for a BE guest or if the host is BE then there may be difficult to track down bugs in the execution!`